### PR TITLE
feat: Add support for FP8 MLA on Hopper and Blackwell.

### PR DESCRIPTION
**Pull request by bobboli**
This PR adds FP8 MLA support on Hopper and Blackwell.
- Recipe: per-tensor FP8 e4m3 quantization for Q and latent KV. MLA output is in BF16. Currently no calibration and the quantization scales are simply set to 1.
- Add code for Q and KV cache quantization.
- trtllm-gen based FP8 MLA kernel support for Blackwell.
- [FP8 FlashMLA kernel](https://github.com/deepseek-ai/FlashMLA/issues/56) support for Hopper. Default option for Hopper now.
- (draft) FMHA based FP8 MLA kernel for Hopper, with accuracy issues to be investigated. This PR also includes some refactoring to the FMHA kernel management code.

### CODE REVIEW

File: tensorrt_llm/_torch/modules/attention.py [hunk 1]
```diff
@@ -531,7 +531,9 @@ def forward_context(
         # Concat q(including q_pe), k + k_pe, v together as input_qkv
         input_qkv = torch.cat([q, k, v], dim=-1)
 
-        out_scale = getattr(self.o_proj, "inv_input_scale", None)
+        # out_scale = getattr(self.o_proj, "inv_input_scale", None)
+        out_scale = None  # Currently we use BF16 MHA for context phase
```
Comment (DylanChen-NV) [inline tensorrt_llm/_torch/modules/attention.py:535]: Is this line supposed to be committed？

Comment (bobboli) [inline tensorrt_llm/_torch/modules/attention.py:535]: If out_scale is not None, attentionOp will assume that the attention output type is FP8. Currently we want to keep context MLA, as well as the output of generation MLA, in Bf16.

**Review by Tracin — APPROVED** [review]
